public/components/common/custom-search-bar/custom-search-bar.tsx

```diff
@@ -0,0 +1,190 @@
+
+    useEffect(() => {
+        let filterSubscriber = filterManager.getUpdates$().subscribe(() => {
+            const newFilters = filterManager.getFilters();
+            onFiltersUpdated(newFilters)
+            return () => {
+                filterSubscriber.unsubscribe();
+            };
+        });
+    }, []);
+
+    const onQuerySubmit = (payload: { dateRange: TimeRange, query: Query }) => {
+        const { query, dateRange } = payload;
+        const filters = { query, time: dateRange, filters: filterParams.filters };
+        setFilterParams(filters);
+
+    }
+
+    const onFiltersUpdated = (filters: Filter[]) => {
+        const { query, time } = filterParams;
+        const updatedFilterParams = { query, time, filters };
+        setFilterParams(updatedFilterParams);
+        refreshCustomSelectedFilter()
+    }
+
+    const changeSwitch = () => {
+        setAvancedFiltersState(state => !state);
+    }
+
+
+    const buildCustomFilter = (isPinned: boolean, index?: any, querySearch?: any, key?: any): Filter => {
+        const meta: FilterMeta = {
+            disabled: false,
+            negate: false,
+            key: key,
+            params: { query: querySearch },
+            alias: null,
+            type: "phrase",
+            index,
+        };
+        const $state: FilterState = {
+            store: isPinned ? FilterStateStore.GLOBAL_STATE : FilterStateStore.APP_STATE,
+        };
+        const query = {
+            match_phrase: {
+                [key]: {
+                    query: querySearch
+                }
+            }
+        }
+
+        return { meta, $state, query };
+    };
+
+    const setKibanaFilters = (values: any[]) => {
+        setLoading(true)
+        const newFilters = []
+        const currentFilters = filterManager.getFilters().filter(item => item.meta.key != values[0].value)
+        filterManager.removeAll()
+        filterManager.addFilters(currentFilters)
+        values.forEach(element => {
+            const customFilter = buildCustomFilter(false, indexPattern.title, element.label, element.value)
+            newFilters.push(customFilter);
+        });
+        filterManager.addFilters(newFilters)
+    }
+
+    const refreshCustomSelectedFilter = () => {
+        setSelectedOptions(defaultSelectedOptions)
+        const filters = filterManager.getFilters()
+        const filterCustom = filters.map(item => {
+            return {
+                value: item.meta.key,
+                label: item.meta.params.query,
+            }
+        }).filter(element => Object.keys(selectedOptions).includes(element.value))
+
+        if (filterCustom.length != 0) {
+            filterCustom.forEach(item => {
+                setSelectedOptions(prevState => ({
+                    ...prevState,
+                    [item.value]: [...prevState[item.value], item],
+                }));
+
+            })
+        }
+        setLoading(false)
```

Comment:
`setLoading` is set to false after another synchronous function. I am not sure when the wait is supposed to happen.

Reply:
`setLoading` is obsolete, it was my mistake. now i upload the change